### ISSUE TITLE
Why Precision of SparkDl4jMultiLayer model  is NaN running on spark cluster?

### ISSUE THREAD

**Issue by ustbxyls** [deeplearning4j/deeplearning4j#5647]
I have trained a SparkDl4jMultiLayer, but I got some weird results of evaluation.
when I run this net in my IDE idea, i got this 
![image](https://user-images.githubusercontent.com/4646748/41591764-3c7e3b46-73ed-11e8-8e26-767e9876b550.png)

buta, when I run this net on spark cluster, i got this 
![image](https://user-images.githubusercontent.com/4646748/41591846-72aacefa-73ed-11e8-94eb-2a843fec6dfc.png)

here is my code ：
 MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
                .seed(seed)
                .updater(new Nesterovs(learningRate, 0.9))
                .list()
                .layer(0, new DenseLayer.Builder().nIn(numInputs).nOut(numHiddenNodes)
                        .weightInit(WeightInit.XAVIER)
                        .activation(Activation.RELU)
                        .build())
                .layer(1, new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                        .weightInit(WeightInit.XAVIER)
                        .activation(Activation.SOFTMAX)
                        .nIn(numHiddenNodes).nOut(numOutputs).build())
                .pretrain(false).backprop(true).build();

        TrainingMaster tm = new ParameterAveragingTrainingMaster.Builder(batchSizePerWorker)
                .averagingFrequency(5)
                .workerPrefetchNumBatches(2)
                .batchSizePerWorker(batchSizePerWorker)
                .build();

        //Create the Spark network
        SparkDl4jMultiLayer sparkNet = new SparkDl4jMultiLayer(sc, conf, tm);

        for ( int i = 0; i <numEpochs; i++) {
            sparkNet.fit(trainData);
            log.info("Completed Epoch {}", i);
            System.out.println("Completed Epoch : " + i);
        }

        System.out.println("Evaluate model....");
        Evaluation eval = sparkNet.doEvaluation(testData, batchSize, new Evaluation(numOutputs))[0];
        System.out.println(eval.stats());`

**Comment by AlexDBlack** [comment]
OK, so this was a know problem with 0.9.1. Upgrade to 1.0.0-beta.
Also don't mix versions like that: ND4J, DL4J and DataVec should all be the same version.
